core_backend/app/languages/routers.py

```diff
@@ -0,0 +1,182 @@
+from typing import Annotated, List
+
+from fastapi import APIRouter, Depends
+from fastapi.exceptions import HTTPException
+from sqlalchemy.ext.asyncio import AsyncSession
+
+from ..auth.dependencies import get_current_fullaccess_user, get_current_readonly_user
+from ..auth.schemas import AuthenticatedUser
+from ..database import get_async_session
+from ..utils import setup_logger
+from .models import (
+    LanguageDB,
+    delete_language_from_db,
+    get_default_language_from_db,
+    get_language_from_db,
+    get_list_of_languages_from_db,
+    is_language_name_unique,
+    save_language_to_db,
+    update_language_in_db,
+)
+from .schemas import LanguageBase, LanguageRetrieve
+
+router = APIRouter(prefix="/languages")
+logger = setup_logger()
+
+
+@router.post("/", response_model=LanguageRetrieve)
```

Comment:
Made changes for the endpoint as per [this thread](https://idinsight.slack.com/archives/C05RSDLN1GA/p1710490893253359)